sfge/src/main/java/com/salesforce/graph/ApexPath.java

```diff
@@ -193,6 +186,38 @@ public ApexPath deepClone() {
         return new ApexPath(this);
     }
 
+    /**
+     * Get every subpath contained within this path.
+     *
+     * @param deep - Also include subpaths of subpaths
+     */
+    public List<ApexPath> getAllSubpaths(boolean deep) {
```

Comment:
@rmohan20 , we discussed moving this to a Util class, but I feel like it's better off here, since it doesn't overly expose the internals of ApexPath. Thoughts?

Reply:
Agree, looks like it depends heavily on the contents of the ApexPath instance. Wonder if there's a way to control access to this from only the UnusedMethodRule. I'm a little nervous about the recursion getting unintentionally invoked from some other place.